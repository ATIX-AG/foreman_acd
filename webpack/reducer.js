import { combineReducers } from 'redux';
import {
  cloneDeep,
  findIndex,
  findLastIndex,
} from 'lodash';

import parameterSelectionParameters from './components/ParameterSelection/ParameterSelectionReducer';
import applicationDefinitionConf from './components/ApplicationDefinition/ApplicationDefinitionReducer';
import applicationInstanceConf from './components/ApplicationInstance/ApplicationInstanceReducer';

import {
  APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationDefinition/ApplicationDefinitionConstants';

import {
  APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationInstance/ApplicationInstanceConstants';

const rootReducer = (state = {}, action: Action) => {

  let param_state = parameterSelectionParameters(state.parameterSelectionParameters, action);

  if (action.type == APPLICATION_DEFINITION_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.serviceParameterSelection = param_state.parameters;
  }
  let app_def_state = applicationDefinitionConf(state.applicationDefinitionConf, action);

  if (action.type == APPLICATION_INSTANCE_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.hostParameterSelection = param_state.parameters;
  }
  let app_ins_state = applicationInstanceConf(state.applicationInstanceConf, action);

  return {
    applicationDefinitionConf: app_def_state,
    applicationInstanceConf: app_ins_state,
    parameterSelectionParameters: param_state,
  };
};

export default rootReducer;
