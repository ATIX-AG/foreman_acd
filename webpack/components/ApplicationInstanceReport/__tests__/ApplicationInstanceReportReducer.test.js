import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import reducer, { initialState } from '../ApplicationInstanceReportReducer';

import {
  successState,
  initApplicationInstanceReportPayload,
  setActiveHostPayload,
} from '../__fixtures__/applicationInstanceReportReducer.fixtures';

import {
  APPLICATION_INSTANCE_REPORT_INIT,
  APPLICATION_INSTANCE_REPORT_SET_ACTIVE_HOST,
} from '../ApplicationInstanceReportConstants';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },

  'should initialize component': {
    state: initialState,
    action: {
      type: APPLICATION_INSTANCE_REPORT_INIT,
      payload: initApplicationInstanceReportPayload,
    },
  },
  'should set active host': {
    state: successState,
    action: {
      type: APPLICATION_INSTANCE_REPORT_SET_ACTIVE_HOST,
      payload: setActiveHostPayload,
    },
  },
};

describe('ApplicationInstanceReport', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
