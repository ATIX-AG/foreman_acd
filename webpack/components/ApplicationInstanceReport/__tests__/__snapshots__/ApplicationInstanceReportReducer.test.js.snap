// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationInstanceReport should initialize component 1`] = `
Object {
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "hosts": Array [
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-db-1.deploy3.dev.atix",
      "id": 8,
      "name": "great-web-app-db-1",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-db-1.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542728.8466988,
          "id": "Set up compute instance great-web-app-db-1.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-db-1.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.190Z",
        },
        Object {
          "created": 1615542728.8474095,
          "id": "Query instance details for great-web-app-db-1.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-db-1.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.385Z",
        },
        Object {
          "created": 1615542728.0081317,
          "id": "dhcp_create_192.168.171.103",
          "name": "Create DHCP Settings for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.584Z",
        },
        Object {
          "created": 1615542728.04177,
          "id": "Create IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.679Z",
        },
        Object {
          "created": 1615542728.0424814,
          "id": "Create Reverse IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.798Z",
        },
        Object {
          "created": 1615542728.0881674,
          "id": "Deploy TFTP PXELinux config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.178Z",
        },
        Object {
          "created": 1615542728.0883749,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.615Z",
        },
        Object {
          "created": 1615542728.088634,
          "id": "Deploy TFTP PXEGrub config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.642Z",
        },
        Object {
          "created": 1615542728.0888274,
          "id": "Deploy TFTP iPXE config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.874Z",
        },
        Object {
          "created": 1615542728.0894718,
          "id": "Fetch TFTP boot files for great-web-app-db-1.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-db-1.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.981Z",
        },
        Object {
          "created": 1615542728.8482428,
          "id": "Power up compute instance great-web-app-db-1.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-db-1.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:12.950Z",
        },
        Object {
          "created": 1615542728.8508224,
          "id": "Cleanup PuppetCA certificates for great-web-app-db-1.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-db-1.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:13.390Z",
        },
        Object {
          "created": 1615542728.8514216,
          "id": "Enable PuppetCA autosigning for great-web-app-db-1.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-db-1.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:13.442Z",
        },
      ],
    },
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-web-1.deploy3.dev.atix",
      "id": 9,
      "name": "great-web-app-web-1",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-web-1.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542733.824889,
          "id": "Set up compute instance great-web-app-web-1.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-web-1.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:14.825Z",
        },
        Object {
          "created": 1615542733.825672,
          "id": "Query instance details for great-web-app-web-1.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-web-1.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.007Z",
        },
        Object {
          "created": 1615542733.5710943,
          "id": "dhcp_create_192.168.171.104",
          "name": "Create DHCP Settings for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.181Z",
        },
        Object {
          "created": 1615542733.599793,
          "id": "Create IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.251Z",
        },
        Object {
          "created": 1615542733.600369,
          "id": "Create Reverse IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.328Z",
        },
        Object {
          "created": 1615542733.6356068,
          "id": "Deploy TFTP PXELinux config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.681Z",
        },
        Object {
          "created": 1615542733.635817,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.152Z",
        },
        Object {
          "created": 1615542733.6360064,
          "id": "Deploy TFTP PXEGrub config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.183Z",
        },
        Object {
          "created": 1615542733.6362345,
          "id": "Deploy TFTP iPXE config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.417Z",
        },
        Object {
          "created": 1615542733.6365006,
          "id": "Fetch TFTP boot files for great-web-app-web-1.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-web-1.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.523Z",
        },
        Object {
          "created": 1615542733.8267431,
          "id": "Power up compute instance great-web-app-web-1.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-web-1.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.533Z",
        },
        Object {
          "created": 1615542733.8292165,
          "id": "Cleanup PuppetCA certificates for great-web-app-web-1.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-web-1.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.944Z",
        },
        Object {
          "created": 1615542733.8294067,
          "id": "Enable PuppetCA autosigning for great-web-app-web-1.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-web-1.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.995Z",
        },
      ],
    },
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-web-2.deploy3.dev.atix",
      "id": 10,
      "name": "great-web-app-web-2",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-web-2.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542738.3446305,
          "id": "Set up compute instance great-web-app-web-2.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-web-2.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.221Z",
        },
        Object {
          "created": 1615542738.3449144,
          "id": "Query instance details for great-web-app-web-2.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-web-2.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.419Z",
        },
        Object {
          "created": 1615542738.1321318,
          "id": "dhcp_create_192.168.171.105",
          "name": "Create DHCP Settings for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.668Z",
        },
        Object {
          "created": 1615542738.1576574,
          "id": "Create IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.749Z",
        },
        Object {
          "created": 1615542738.1582346,
          "id": "Create Reverse IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.859Z",
        },
        Object {
          "created": 1615542738.1885784,
          "id": "Deploy TFTP PXELinux config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.221Z",
        },
        Object {
          "created": 1615542738.1888642,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.604Z",
        },
        Object {
          "created": 1615542738.1890883,
          "id": "Deploy TFTP PXEGrub config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.636Z",
        },
        Object {
          "created": 1615542738.1892834,
          "id": "Deploy TFTP iPXE config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.895Z",
        },
        Object {
          "created": 1615542738.1898031,
          "id": "Fetch TFTP boot files for great-web-app-web-2.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-web-2.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.992Z",
        },
        Object {
          "created": 1615542738.3453662,
          "id": "Power up compute instance great-web-app-web-2.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-web-2.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:21.864Z",
        },
        Object {
          "created": 1615542738.34748,
          "id": "Cleanup PuppetCA certificates for great-web-app-web-2.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-web-2.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:22.316Z",
        },
        Object {
          "created": 1615542738.3476808,
          "id": "Enable PuppetCA autosigning for great-web-app-web-2.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-web-2.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:22.368Z",
        },
      ],
    },
  ],
  "name": false,
}
`;

exports[`ApplicationInstanceReport should return initial state 1`] = `
Object {
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "name": false,
}
`;

exports[`ApplicationInstanceReport should set active host 1`] = `
Object {
  "activeHostId": 2,
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "hosts": Array [
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-db-1.deploy3.dev.atix",
      "id": 8,
      "name": "great-web-app-db-1",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-db-1.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542728.8466988,
          "id": "Set up compute instance great-web-app-db-1.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-db-1.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.190Z",
        },
        Object {
          "created": 1615542728.8474095,
          "id": "Query instance details for great-web-app-db-1.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-db-1.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.385Z",
        },
        Object {
          "created": 1615542728.0081317,
          "id": "dhcp_create_192.168.171.103",
          "name": "Create DHCP Settings for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.584Z",
        },
        Object {
          "created": 1615542728.04177,
          "id": "Create IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.679Z",
        },
        Object {
          "created": 1615542728.0424814,
          "id": "Create Reverse IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-db-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:10.798Z",
        },
        Object {
          "created": 1615542728.0881674,
          "id": "Deploy TFTP PXELinux config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.178Z",
        },
        Object {
          "created": 1615542728.0883749,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.615Z",
        },
        Object {
          "created": 1615542728.088634,
          "id": "Deploy TFTP PXEGrub config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.642Z",
        },
        Object {
          "created": 1615542728.0888274,
          "id": "Deploy TFTP iPXE config for great-web-app-db-1.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-db-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.874Z",
        },
        Object {
          "created": 1615542728.0894718,
          "id": "Fetch TFTP boot files for great-web-app-db-1.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-db-1.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:11.981Z",
        },
        Object {
          "created": 1615542728.8482428,
          "id": "Power up compute instance great-web-app-db-1.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-db-1.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:12.950Z",
        },
        Object {
          "created": 1615542728.8508224,
          "id": "Cleanup PuppetCA certificates for great-web-app-db-1.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-db-1.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:13.390Z",
        },
        Object {
          "created": 1615542728.8514216,
          "id": "Enable PuppetCA autosigning for great-web-app-db-1.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-db-1.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:13.442Z",
        },
      ],
    },
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-web-1.deploy3.dev.atix",
      "id": 9,
      "name": "great-web-app-web-1",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-web-1.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542733.824889,
          "id": "Set up compute instance great-web-app-web-1.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-web-1.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:14.825Z",
        },
        Object {
          "created": 1615542733.825672,
          "id": "Query instance details for great-web-app-web-1.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-web-1.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.007Z",
        },
        Object {
          "created": 1615542733.5710943,
          "id": "dhcp_create_192.168.171.104",
          "name": "Create DHCP Settings for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.181Z",
        },
        Object {
          "created": 1615542733.599793,
          "id": "Create IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.251Z",
        },
        Object {
          "created": 1615542733.600369,
          "id": "Create Reverse IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-web-1.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.328Z",
        },
        Object {
          "created": 1615542733.6356068,
          "id": "Deploy TFTP PXELinux config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:15.681Z",
        },
        Object {
          "created": 1615542733.635817,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.152Z",
        },
        Object {
          "created": 1615542733.6360064,
          "id": "Deploy TFTP PXEGrub config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.183Z",
        },
        Object {
          "created": 1615542733.6362345,
          "id": "Deploy TFTP iPXE config for great-web-app-web-1.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-web-1.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.417Z",
        },
        Object {
          "created": 1615542733.6365006,
          "id": "Fetch TFTP boot files for great-web-app-web-1.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-web-1.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:16.523Z",
        },
        Object {
          "created": 1615542733.8267431,
          "id": "Power up compute instance great-web-app-web-1.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-web-1.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.533Z",
        },
        Object {
          "created": 1615542733.8292165,
          "id": "Cleanup PuppetCA certificates for great-web-app-web-1.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-web-1.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.944Z",
        },
        Object {
          "created": 1615542733.8294067,
          "id": "Enable PuppetCA autosigning for great-web-app-web-1.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-web-1.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:17.995Z",
        },
      ],
    },
    Object {
      "build": false,
      "hostUrl": "/hosts/great-web-app-web-2.deploy3.dev.atix",
      "id": 10,
      "name": "great-web-app-web-2",
      "powerStatusUrl": "/api/v2/hosts/great-web-app-web-2.deploy3.dev.atix/power",
      "progress_report": Array [
        Object {
          "created": 1615542738.3446305,
          "id": "Set up compute instance great-web-app-web-2.deploy3.dev.atix",
          "name": "Set up compute instance great-web-app-web-2.deploy3.dev.atix",
          "priority": 3,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.221Z",
        },
        Object {
          "created": 1615542738.3449144,
          "id": "Query instance details for great-web-app-web-2.deploy3.dev.atix",
          "name": "Query instance details for great-web-app-web-2.deploy3.dev.atix",
          "priority": 5,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.419Z",
        },
        Object {
          "created": 1615542738.1321318,
          "id": "dhcp_create_192.168.171.105",
          "name": "Create DHCP Settings for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.668Z",
        },
        Object {
          "created": 1615542738.1576574,
          "id": "Create IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "name": "Create IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.749Z",
        },
        Object {
          "created": 1615542738.1582346,
          "id": "Create Reverse IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "name": "Create Reverse IPv4 DNS record for great-web-app-web-2.deploy3.dev.atix",
          "priority": 10,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:19.859Z",
        },
        Object {
          "created": 1615542738.1885784,
          "id": "Deploy TFTP PXELinux config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXELinux config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.221Z",
        },
        Object {
          "created": 1615542738.1888642,
          "id": "Deploy TFTP PXEGrub2 config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub2 config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.604Z",
        },
        Object {
          "created": 1615542738.1890883,
          "id": "Deploy TFTP PXEGrub config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP PXEGrub config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.636Z",
        },
        Object {
          "created": 1615542738.1892834,
          "id": "Deploy TFTP iPXE config for great-web-app-web-2.deploy3.dev.atix",
          "name": "Deploy TFTP iPXE config for great-web-app-web-2.deploy3.dev.atix",
          "priority": 20,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.895Z",
        },
        Object {
          "created": 1615542738.1898031,
          "id": "Fetch TFTP boot files for great-web-app-web-2.deploy3.dev.atix",
          "name": "Fetch TFTP boot files for great-web-app-web-2.deploy3.dev.atix",
          "priority": 25,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:20.992Z",
        },
        Object {
          "created": 1615542738.3453662,
          "id": "Power up compute instance great-web-app-web-2.deploy3.dev.atix",
          "name": "Power up compute instance great-web-app-web-2.deploy3.dev.atix",
          "priority": 1000,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:21.864Z",
        },
        Object {
          "created": 1615542738.34748,
          "id": "Cleanup PuppetCA certificates for great-web-app-web-2.deploy3.dev.atix",
          "name": "Cleanup PuppetCA certificates for great-web-app-web-2.deploy3.dev.atix",
          "priority": 51,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:22.316Z",
        },
        Object {
          "created": 1615542738.3476808,
          "id": "Enable PuppetCA autosigning for great-web-app-web-2.deploy3.dev.atix",
          "name": "Enable PuppetCA autosigning for great-web-app-web-2.deploy3.dev.atix",
          "priority": 55,
          "status": "completed",
          "timestamp": "2021-03-12T09:52:22.368Z",
        },
      ],
    },
  ],
  "name": false,
}
`;
